lib/sensors/src/accelerometer.rs

```diff
@@ -0,0 +1,153 @@
+use hyped_io::i2c::{HypedI2c, I2cError};
+
+/// Accelerometer implements the logic to read the temperature from the LIS2DS12 accelerometer
+/// using the peripheral provided by the HypedI2c trait.
+///
+/// Based on last year's implementation (might need to change this later),
+/// The accelerometer is configured to a sampling rate of 200Hz with high performance mode and continuous update.
+/// The full scale of the accelerometer (+-16g) is used.
+///  
+/// The acceleration value for each axis is read from the sensor and converted to a floating point value in gs. 
+/// 
+/// Data sheet: https://www.st.com/resource/en/datasheet/lis2ds12.pdf
+/// Application notes: https://www.st.com/resource/en/application_note/an4748-lis2ds12-alwayson-3axis-accelerometer-stmicroelectronics.pdf
+pub struct Accelerometer<T: HypedI2c> {
+    i2c: T,
+    device_address: u8,
+}
+
+impl<T: HypedI2c> Accelerometer<T> {
+    /// Create a new instance of the accelerometer and attempt to configure it
+    pub fn new(mut i2c: T, device_address: AccelerometerAddresses) -> Result<Self, AccelerometerError> {
+        let device_address = device_address as u8;
+        match i2c.write_byte_to_register(device_address, LIS2DS12_CTRL1_ADDRESS, LIS2DS12_CTRL1_VALUE) {
+            Err(e) => return Err(AccelerometerError::I2cError(e)),
+            Ok(p) =>  p
+        };
+        match i2c.write_byte_to_register(device_address, LIS2DS12_CTRL2_ADDRESS, LIS2DS12_CTRL2_VALUE) {
+            Err(e) => return Err(AccelerometerError::I2cError(e)),
+            Ok(p) =>  p
+        };
+        match i2c.write_byte_to_register(device_address, LIS2DS12_FIFO_CTRL_ADDRESS, LIS2DS12_FIFO_CTRL_VALUE) {
+            Err(e) => return Err(AccelerometerError::I2cError(e)),
+            Ok(p) =>  p
+        };
+
+        // Return Self only if all values are written successfully
+        Ok(Self{i2c, device_address})
+    }
+
+    /// Read the acceleration for each axis and return them as a floating point value in gs.
+    pub fn read(&mut self) -> Option<(f32, f32, f32)> {
```

Comment:
is it preferred to return these as a tuple or have named fields in a struct to keep it clear which is x/y/z?

Reply:
Yeah I think a struct is better because it's more explicit